eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/EnactSpec.hs

```diff
@@ -79,145 +100,343 @@ spec =
+
+    proposalsBeforeVotes <- getsNES $ newEpochStateGovStateL . proposalsGovStateL
+    pulserBeforeVotes <- getsNES newEpochStateDRepPulsingStateL
+
+    submitYesVote_ (DRepVoter dRep) govActionId
+    submitYesVote_ (CommitteeVoter committeeMember) govActionId
+
+    proposalsAfterVotes <- getsNES $ newEpochStateGovStateL . proposalsGovStateL
+    pulserAfterVotes <- getsNES newEpochStateDRepPulsingStateL
+
+    impAnn "Votes are recorded in the proposals" $ do
+      let proposalsWithVotes =
+            proposalsAddVote
+              (CommitteeVoter committeeMember)
+              VoteYes
+              govActionId
+              ( proposalsAddVote
+                  (DRepVoter dRep)
+                  VoteYes
+                  govActionId
+                  proposalsBeforeVotes
+              )
+      proposalsAfterVotes `shouldBe` proposalsWithVotes
+
+    impAnn "Pulser has not changed" $
+      pulserAfterVotes `shouldBe` pulserBeforeVotes
+
+    passEpoch
+
+    impAnn "New constitution is not enacted after one epoch" $ do
+      constitutionAfterOneEpoch <- getsNES $ newEpochStateGovStateL . constitutionGovStateL
+      constitutionAfterOneEpoch `shouldBe` def
+
+    impAnn "Pulser should reflect the constitution to be enacted" $ do
+      pulser <- getsNES newEpochStateDRepPulsingStateL
+      let ratifyState = extractDRepPulsingState pulser
+      gasId <$> rsEnacted ratifyState `shouldBe` govActionId Seq.:<| Seq.Empty
+      rsEnactState ratifyState ^. ensConstitutionL `shouldBe` constitution
+
+    passEpoch
 
-      impAnn "New constitution is not enacted after one epoch" $ do
-        constitutionAfterOneEpoch <- getsNES $ newEpochStateGovStateL . constitutionGovStateL
-        constitutionAfterOneEpoch `shouldBe` def
+    impAnn "Constitution is enacted after two epochs" $ do
+      curConstitution <- getsNES $ newEpochStateGovStateL . constitutionGovStateL
+      curConstitution `shouldBe` constitution
 
-      impAnn "Pulser should reflect the constitution to be enacted" $ do
-        ConwayGovState _ _ _ _ _ pulser <- getsNES newEpochStateGovStateL
-        let ratifyState = extractDRepPulsingState pulser
-        gasId <$> rsEnacted ratifyState `shouldBe` govActionId Seq.:<| Seq.Empty
-        rsEnactState ratifyState ^. ensConstitutionL `shouldBe` constitution
+    impAnn "Pulser is reset" $ do
+      pulser <- getsNES newEpochStateDRepPulsingStateL
+      let pulserRatifyState = extractDRepPulsingState pulser
+      rsEnacted pulserRatifyState `shouldBe` Seq.empty
+      enactState <- getEnactState
+      rsEnactState pulserRatifyState `shouldBe` enactState
 
-      passEpoch
+actionPrioritySpec ::
+  forall era.
+  ConwayEraImp era =>
+  SpecWith (ImpTestState era)
+actionPrioritySpec =
+  describe "Competing proposals ratified in the same epoch" $ do
+    it
+      "The first submitted proposal of higher priority wins, even when submitted after a lower priority one"
+      $ do
```

Comment:
Test names should be succinct and unique, so they can be easily found in the codebase. They don't need to describe the full behavior of the test, that can be explained with a comment above the test case.
```suggestion
    -- The first submitted proposal of higher priority wins, even when 
    -- submitted after a lower priority one
    it "Proposal of higher priority wins" $ do
```

Reply:
Just read the resolved suggestion, seems like @aniketd suggested the same thing :slightly_smiling_face: 

Notice that allows putting both `it` and `do` on the same line, making the code more readable too.